flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitter.java

```diff
@@ -240,12 +256,22 @@ public void processElement(StreamRecord<DataFile> element) {
   }
 
   @Override
-  public void endInput() {
+  public void endInput() throws IOException {
     // Flush the buffered data files into 'dataFilesPerCheckpoint' firstly.
-    dataFilesPerCheckpoint.put(Long.MAX_VALUE, ImmutableList.copyOf(dataFilesOfCurrentCheckpoint));
+    long currentCheckpointId = Long.MAX_VALUE;
+    dataFilesPerCheckpoint.put(currentCheckpointId, writeToManifest(currentCheckpointId));
     dataFilesOfCurrentCheckpoint.clear();
 
-    commitUpToCheckpoint(dataFilesPerCheckpoint, flinkJobId, Long.MAX_VALUE);
+    commitUpToCheckpoint(dataFilesPerCheckpoint, flinkJobId, currentCheckpointId);
+  }
+
+  /**
+   * Write all the complete data files to a newly created manifest file and return the manifest's avro serialized bytes.
+   */
+  private Byte[] writeToManifest(long checkpointId) throws IOException {
```

Comment:
Why is this `Byte[]` instead of `byte[]`?

Reply:
I thought that flink state backend could only support `Byte[]`  type,  but after skimmed the flink code again, I found that we have `PrimitiveArrayTypeInfo.BYTE_PRIMITIVE_ARRAY_TYPE_INFO`  to support `byte[]`.  So here we should use the `byte[]` directly.